hal/rtc_api.h

```diff
@@ -30,37 +30,147 @@ extern "C" {
 #endif
 
 /**
- * \defgroup hal_rtc RTC hal functions
+ * \defgroup hal_rtc RTC hal
+ *
+ * The RTC hal provides a low level interface to the Real Time Counter (RTC) of a
+ * target.
+ *
+ * # Defined behavior
+ * * The function ::rtc_init is safe to call repeatedly - Verified by test ::rtc_init_test
+ * * RTC accuracy is at least 10% - Verified by test ::rtc_accuracy_test
+ * * Init/free doesn't stop RTC from counting - Verified by test ::rtc_persist_test
+ * * Software reset doesn't stop RTC from counting - Verified by ::rtc_reset_test
+ * * Sleep modes don't stop RTC from counting - Verified by ::rtc_sleep_test
+ * * Shutdown mode doesn't stop RTC from counting - Not verified
+ * * The functions ::rtc_write/::rtc_read provides availability to set/get RTC time - Verified by ::rtc_write_read_test
+ * * The functions ::rtc_isenabled returns 1 if the RTC is counting and the time has been set, 0 otherwise - Verified by ::rtc_enabled_test
+ *
+ * # Undefined behavior
+ * * Calling any function other than ::rtc_init before the initialization of the RTC
+ *
+ * # Potential bugs
+ * * Incorrect overflow handling - Verified by ::rtc_range_test
+ * * Glitches due to ripple counter - Verified by ::rtc_glitch_test
+ *
+ * @see hal_rtc_tests
+ *
  * @{
  */
 
+/**
+ * \defgroup hal_rtc_tests RTC hal tests
+ * The RTC test validate proper implementation of the RTC hal.
+ *
+ * To run the RTC hal tests use the command:
+ *
+ *     mbed test -t <toolchain> -m <target> -n tests-mbed_hal-rtc*
+ */
+
+
 /** Initialize the RTC peripheral
  *
+ * Powerup the RTC in perpetration for access. This function must be called
+ * before any other RTC functions ares called. This does not change the state
```

Comment:
Do we test that that actually does not change the state? Eg does not reset the counter?

Reply:
`rtc_persist_test()` test case verifies if counter value is valid after free and re-initialisation (still counts).